### ISSUE TITLE
Is Profile broken on Windows?

### ISSUE THREAD

**Issue by PetrKryslUCSD** [JuliaLang/julia#35719]
````
Julia Version 1.4.0                                                              
Commit b8e9a9ecc6 (2020-03-21 16:36 UTC)                                         
Platform Info:                                                                   
  OS: Windows (x86_64-w64-mingw32)                                               
  CPU: Intel(R) Core(TM) i7-2670QM CPU @ 2.20GHz                                 
  WORD_SIZE: 64                                                                  
  LIBM: libopenlibm                                                              
  LLVM: libLLVM-8.0.1 (ORCJIT, sandybridge)                                      
Environment:                                                                     
  JULIA_NUM_THREADS = 4   
````

````
julia> include("C:\\Users\\PKrysl\\Documents\\work\\FEMist.jl\\examples\\heat\\heat_poisson.jl");                                                                 
WARNING: replacing module heat_poisson.                                          
Overhead ╎ [+additional indent] Count File:Line; Function                        
=========================================================                        
  ╎521 @Base\task.jl:358; (::REPL.var"#26#27"{REPL.REPLBack...                   
ERROR: LoadError: IOError: stat: unknown error (UNKNOWN) for file "D:\\buildbot\\worker\\package_win64\\build\\usr\\share\\julia\\stdlib\\v1.4\\REPL\\src\\REPL.jl"                                                                                
Stacktrace:                                                                      
 [1] stat(::String) at .\stat.jl:69                                              
 [2] ispath at .\stat.jl:311 [inlined]                                           
 [3] (::Profile.var"#8#9"{Symbol})() at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Profile\src\Profile.jl:255                             
 [4] get!(::Profile.var"#8#9"{Symbol}, ::Dict{Symbol,String}, ::Symbol) at .\dict.jl:452                                                                          
 [5] short_path at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Profile\src\Profile.jl:252 [inlined]                                        
 [6] tree_format(::Array{Profile.StackFrameTree{Base.StackTraces.StackFrame},1}, 
::Int64, ::Int64, ::NamedTuple{(:count, :count_flat, :overhead, :max_recur),NTuple{4,Int64}}, ::Dict{Symbol,String}, ::Bool) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Profile\src\Profile.jl:580                      
 [7] print_tree(::Base.TTY, ::Profile.StackFrameTree{Base.StackTraces.StackFrame}, ::Int64, ::Profile.ProfileFormat) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Profile\src\Profile.jl:748                              
 [8] tree(::Base.TTY, ::Array{UInt64,1}, ::Dict{UInt64,Array{Base.StackTraces.StackFrame,1}}, ::Int64, ::Profile.ProfileFormat) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Profile\src\Profile.jl:785                   
 [9] print(::Base.TTY, ::Array{UInt64,1}, ::Dict{UInt64,Array{Base.StackTraces.StackFrame,1}}, ::Profile.ProfileFormat, ::Symbol) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Profile\src\Profile.jl:164                 
 [10] #print#4 at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Profile\src\Profile.jl:148 [inlined]                                         
 [11] print(::Base.TTY, ::Array{UInt64,1}, ::Dict{UInt64,Array{Base.StackTraces.StackFrame,1}}) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Profile\src\Profile.jl:148                                                   
 [12] print(::Array{UInt64,1}, ::Dict{UInt64,Array{Base.StackTraces.StackFrame,1}}; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) 
at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Profile\src\Profile.jl:182                                                                  
 [13] print at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.4\Profile\src\Profile.jl:182 [inlined] (repeats 3 times)                          
 [14] assembleK(::FEMist.IntegDomains.IntegDomain) at C:\Users\PKrysl\Documents\work\FEMist.jl\examples\heat\heat_poisson.jl:195                                  
 [15] run() at C:\Users\PKrysl\Documents\work\FEMist.jl\examples\heat\heat_poisson.jl:210                                                                         
 [16] top-level scope at C:\Users\PKrysl\Documents\work\FEMist.jl\examples\heat\heat_poisson.jl:215                                                               
 [17] include(::String) at .\client.jl:439                                       
 [18] top-level scope at REPL[2]:1                                               
in expression starting at C:\Users\PKrysl\Documents\work\FEMist.jl\examples\heat\heat_poisson.jl:215      
````

**Comment by PetrKryslUCSD** [comment]
Similar errors occur with 1.4.1 and 1.5.

**Comment by kimikage** [comment]
I have the same problem on my Windows machine. As the stack trace shows, this is a problem with `print()` (specifically, `short_path()` in `Profile`, or `stat()` in `Base`, or "libuv"), not the profiling mechanism itself,

```julia
julia> stat("/cygdrive/d/")
StatStruct(mode=0o000000, size=0)

julia> stat("D:\\") # drive "D" does not exist
ERROR: IOError: stat: unknown error (UNKNOWN) for file "D:\\"
```

I think that what we should do in `Profile` is to enclose `ispath()` with a `try` block as Pkg does, for now. :confused:

**Comment by ueliwechsler** [comment]
I got a similar error with `stat` while parsing a file with `JSON.parsefile`, but I am not yet sure what exactly is causing this error. 
```julia
julia> versioninfo()
Julia Version 1.4.2
Commit 44fa15b150* (2020-05-23 18:35 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Xeon(R) Platinum 8168 CPU @ 2.70GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-8.0.1 (ORCJIT, skylake)
```

```julia
System_internal_error Base.IOError("stat: unknown error (UNKNOWN) for file \"I:\\\\NonProd\\\\Dyson\\\\DJSON\\\\config.json\"", -4094)
    stat(::String) at stat.jl:69
    filesize at stat.jl:311 [inlined]
    #parsefile#3 at Parser.jl:521 [inlined]
    parsefile at Parser.jl:521 [inlined]
    _config_params(::String, ::Vararg{String,N} where N) at DJSON.jl:54
    load_portfolio_service(::DJSON.MicroServices.MicroServiceController) at MicroServices.jl:26
    _apply_action(::Bukdu.Route, ::Bukdu.Plug.Conn) at routes.jl:46
    #10 at routes.jl:78 [inlined]
    _catch_internal_error(::Bukdu.var"#10#11"{Bukdu.Route,Bukdu.Plug.Conn}, ::Bukdu.Route, ::Bukdu.Plug.Conn) at routes.jl:64
    _proc_request at routes.jl:76 [inlined]
    request_handler(::Bukdu.Route, ::Bukdu.Plug.Conn) at routes.jl:102
    handle_request(::HTTP.Messages.Request, ::Nothing) at server.jl:26
    handle_stream(::HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}, ::Nothing) at server.jl:34
    handle_stream_without_remote_ip at server.jl:54 [inlined]
    macro expansion at Servers.jl:367 [inlined]
    (::HTTP.Servers.var"#13#14"{typeof(Bukdu.handle_stream_without_remote_ip),HTTP.ConnectionPool.Transaction{Sockets.TCPSocket},HTTP.Streams.Stream{HTTP.Messages.Request,HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}})() at task.jl:358
```

**Comment by mossr** [comment]
Confirmed that this is broken on Windows 10 with Julia v1.5.2:
```julia
Julia Version 1.5.2
Commit 539f3ce943 (2020-09-23 23:17 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, skylake)
```

But seems to be **fixed** in Julia v1.6:
```julia
Julia Version 1.6.0
Commit f9720dc2eb (2021-03-24 12:55 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake)
```
